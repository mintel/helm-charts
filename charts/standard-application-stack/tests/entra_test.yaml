suite: Test Entra
release:
  namespace: test-namespace

tests:
  - it: Renders ALB Role when entra enabled
    template: entra-alb-role.yaml
    set:
      global.name: test-app
      global.clusterEnv: dev
      entra.enabled: true
    asserts:
      - isKind:
          of: Role

  - it: Renders ALB RoleBinding when entra enabled
    template: entra-alb-rolebinding.yaml
    set:
      global.name: test-app
      global.clusterEnv: dev
      entra.enabled: true
    asserts:
      - isKind:
          of: RoleBinding

  - it: Renders PasswordCredential when entra enabled
    template: entra-password-credentials.yaml
    set:
      global.name: test-app
      global.clusterEnv: dev
      entra.enabled: true
      entra.uniqueName: test-entra-app
    asserts:
      - isKind:
          of: PasswordCredential

  - it: Does not render Application when not prod
    template: entra-application.yaml
    set:
      global.name: test-app
      global.clusterEnv: dev
      entra.enabled: true
      entra.uniqueName: test-entra-app
      entra.displayName: Test Entra App
      entra.description: Test description
    asserts:
      - hasDocuments:
          count: 0

  - it: Renders Application when entra enabled and prod
    template: entra-application.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      entra.enabled: true
      entra.uniqueName: test-entra-app
      entra.displayName: Test Entra App
      entra.description: Test description
    asserts:
      - isKind:
          of: Application

  - it: Does not render ServicePrincipal when not prod
    template: entra-service-principal.yaml
    set:
      global.name: test-app
      global.clusterEnv: dev
      entra.enabled: true
      entra.uniqueName: test-entra-app
      entra.displayName: Test Entra App
      entra.description: Test description
    asserts:
      - hasDocuments:
          count: 0

  - it: Renders ServicePrincipal when entra enabled and prod (default options)
    template: entra-service-principal.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      entra.enabled: true
      entra.uniqueName: test-entra-app
      entra.displayName: Test Entra App
      entra.description: Test description
    asserts:
      - isKind:
          of: ServicePrincipal
      - equal:
          path: spec.appRoleAssignmentRequired
          value: true
      - equal:
          path: spec.featureTags.visibleToUsers
          value: true

  - it: Renders ServicePrincipal with appRoleAssignmentRequired set to false
    template: entra-service-principal.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      entra.enabled: true
      entra.uniqueName: test-entra-app
      entra.displayName: Test Entra App
      entra.description: Test description
      entra.appRoleAssignmentRequired: false
    asserts:
      - isKind:
          of: ServicePrincipal
      - equal:
          path: spec.appRoleAssignmentRequired
          value: false

  - it: Renders ServicePrincipal with visibleToUsers set to false
    template: entra-service-principal.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      entra.enabled: true
      entra.uniqueName: test-entra-app
      entra.displayName: Test Entra App
      entra.description: Test description
      entra.visibleToUsers: false
    asserts:
      - isKind:
          of: ServicePrincipal
      - equal:
          path: spec.featureTags.visibleToUsers
          value: false

  - it: Renders with optional fields (owners, groupMembershipClaims, redirectURIs, extraResourceAccess) in prod
    template: entra-application.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      entra.enabled: true
      entra.uniqueName: test-entra-app
      entra.displayName: Test Entra App
      entra.description: Test description
      entra.owners:
        - owner-group-id-1
        - owner-group-id-2
      entra.groupMembershipClaims:
        - SecurityGroup
        - All
      entra.redirectURIs:
        - https://example.com/callback
        - https://example.com/other
      entra.extraResourceAccess:
        - id: extra-id-1
          type: Scope
        - id: extra-id-2
          type: Role
    asserts:
      - matchSnapshot: {}
      - hasDocuments:
          count: 1
      - isKind:
          of: Application
      - equal:
          path: spec.owners
          value:
            - group:
                id: owner-group-id-1
            - group:
                id: owner-group-id-2
      - equal:
          path: spec.groupMembershipClaims
          value:
            - SecurityGroup
            - All
      - equal:
          path: spec.web.redirectURIs
          value:
            - https://example.com/callback
            - https://example.com/other
      - equal:
          path: spec.requiredResourceAccess[0].resourceAccess
          value:
            - id: 64a6cdd6-aab1-4aaf-94b8-3cc8405e90d0
              type: Scope
            - id: 37f7f235-527c-4136-accd-4a02d197296e
              type: Scope
            - id: 14dad69e-099b-42c9-810b-d002981feec1
              type: Scope
            - id: extra-id-1
              type: Scope
            - id: extra-id-2
              type: Role

  - it: Does not render ALB Role when createIngressRBAC is false
    template: entra-alb-role.yaml
    set:
      global.name: test-app
      global.clusterEnv: dev
      entra.enabled: true
      entra.createIngressRBAC: false
    asserts:
      - hasDocuments:
          count: 0

  - it: Does not render ALB RoleBinding when createIngressRBAC is false
    template: entra-alb-rolebinding.yaml
    set:
      global.name: test-app
      global.clusterEnv: dev
      entra.enabled: true
      entra.createIngressRBAC: false
    asserts:
      - hasDocuments:
          count: 0

  - it: Does not render ALB Role when entra.enabled is false
    template: entra-alb-role.yaml
    set:
      global.name: test-app
      global.clusterEnv: dev
      entra.enabled: false
      entra.createIngressRBAC: true
    asserts:
      - hasDocuments:
          count: 0

  - it: Does not render ALB RoleBinding when entra.enabled is false
    template: entra-alb-rolebinding.yaml
    set:
      global.name: test-app
      global.clusterEnv: dev
      entra.enabled: false
      entra.createIngressRBAC: true
    asserts:
      - hasDocuments:
          count: 0
