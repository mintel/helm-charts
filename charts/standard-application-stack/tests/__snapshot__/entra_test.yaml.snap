Renders with optional fields (owners, groupMembershipClaims, redirectURIs, extraResourceAccess) in prod:
  1: |
    apiVersion: entraid.mintel.cloud/v1alpha1
    kind: Application
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      description: Test description
      displayName: Test Entra App
      groupMembershipClaims:
        - SecurityGroup
        - All
      owners:
        - group:
            id: owner-group-id-1
        - group:
            id: owner-group-id-2
      requiredResourceAccess:
        - resourceAccess:
            - id: 64a6cdd6-aab1-4aaf-94b8-3cc8405e90d0
              type: Scope
            - id: 37f7f235-527c-4136-accd-4a02d197296e
              type: Scope
            - id: 14dad69e-099b-42c9-810b-d002981feec1
              type: Scope
            - id: extra-id-1
              type: Scope
            - id: extra-id-2
              type: Role
          resourceAppID: 00000003-0000-0000-c000-000000000000
      uniqueName: test-entra-app
      web:
        redirectURIs:
          - https://example.com/callback
          - https://example.com/other
